src/data/fintype/basic.lean

```diff
@@ -663,13 +670,52 @@ by simp [finset.subset_iff, set.subset_def]
   s.to_finset ⊂ t.to_finset ↔ s ⊂ t :=
 by simp only [finset.ssubset_def, to_finset_mono, ssubset_def]
 
+lemma subset_iff_to_finset_subset (s t : set α) [fintype s] [fintype t] :
+  s ⊆ t ↔ s.to_finset ⊆ t.to_finset :=
+by simp only [to_finset_mono]
+
 @[simp] theorem to_finset_disjoint_iff [decidable_eq α] {s t : set α} [fintype s] [fintype t] :
   disjoint s.to_finset t.to_finset ↔ disjoint s t :=
 by simp only [disjoint_iff_disjoint_coe, coe_to_finset]
 
+lemma to_finset_inter {α : Type*} [decidable_eq α] (s t : set α) [fintype (s ∩ t : set α)]
```

Comment:
I feel like we should have a library note explaining this pattern (ie why we include instances which are inferrable from others)

Reply:
I created an issue: #14275